WordPress/Classes/Models/WPAccount+Lookup.swift

```diff
@@ -0,0 +1,157 @@
+import CoreData
+
+public extension WPAccount {
+
+    // MARK: - Relationship Lookups
+
+    /// Is this `WPAccount` object the default WordPress.com account?
+    ///
+    @objc
+    var isDefaultWordPressComAccount: Bool {
+        guard let uuid = UserSettings.defaultDotComUUID, uuid.count > 0 else {
+            return false
+        }
+
+        return self.uuid == uuid
+    }
+
+    /// Does this `WPAccount` object have any associated blogs?
+    ///
+    @objc
+    var hasBlogs: Bool {
+        return !blogs.isEmpty
+    }
+
+    // MARK: - Object Lookups
+
+    /// Returns the default WordPress.com account, if one exists
+    ///
+    /// The default WordPress.com account is the one used for Reader and Notifications.
+    ///
+    static func lookupDefaultWordPressComAccount(in context: NSManagedObjectContext) throws -> WPAccount? {
+        guard let uuid = UserSettings.defaultDotComUUID, uuid.count > 0 else {
+            // No account, or no default account set. Clear the defaults key.
+            UserSettings.defaultDotComUUID = nil
```

Comment:
If the guard fails, presumably this will already be nil? (Or I guess, it could be an empty string)

Reply:
Ah yeah, we can clean that up now. It was a holdover from Obj-C bridging because of:

NSString *string = nil;
XCTAssertFalse([string isEmpty]);
XCTAssertTrue([string length] == 0);
The Swift-based property wrapper will deal with that case for us now, so we just need to handle the empty case, and !isEmpty is much more clear there for sure 🙂

Adjusted in f6ce941